app/src/community/res/layout/development_activity_login.xml

```diff
@@ -1,72 +1,89 @@
 
             <ImageView
                 android:layout_width="74dp"
                 android:layout_height="74dp"
                 android:layout_gravity="center_horizontal"
                 android:contentDescription="@string/video_logo"
-                android:src="@drawable/video_logo_splash"/>
+                android:src="@drawable/video_logo_splash" />
 
-            <EditText
-                android:id="@+id/name_edittext"
-                android:layout_marginTop="@dimen/username_margin_top"
-                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+            <com.google.android.material.textfield.TextInputLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
                 android:layout_marginStart="@dimen/activity_horizontal_margin"
-                android:layout_marginRight="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="@dimen/username_margin_top"
                 android:layout_marginEnd="@dimen/activity_horizontal_margin"
-                android:layout_marginBottom="@dimen/activity_horizontal_margin"
+                android:layout_marginRight="@dimen/activity_horizontal_margin"
+                android:layout_marginBottom="@dimen/activity_horizontal_margin">
+
+                <com.google.android.material.textfield.TextInputEditText
+                    android:id="@+id/community_login_screen_name_edittext"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:hint="@string/login_screen_name"
+                    android:inputType="textPersonName"/>
+            </com.google.android.material.textfield.TextInputLayout>
+
+            <com.google.android.material.textfield.TextInputLayout
                 android:layout_width="match_parent"
                 android:layout_height="wrap_content"
-                android:inputType="textPersonName"
-                android:hint="@string/name"/>
+                android:layout_marginStart="@dimen/activity_horizontal_margin"
+                android:layout_marginLeft="@dimen/activity_horizontal_margin"
+                android:layout_marginTop="8dp"
+                android:layout_marginEnd="@dimen/activity_horizontal_margin"
+                android:layout_marginRight="@dimen/activity_horizontal_margin"
+                android:layout_marginBottom="@dimen/activity_horizontal_margin">
+
+                <com.google.android.material.textfield.TextInputEditText
+                    android:id="@+id/community_login_screen_app_code_edittext"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:hint="@string/login_screen_app_code"
+                    android:inputType="textPersonName"/>
```

Comment:
Let's make the `inputType=numberPassword`

Reply:
I thought the same thing at first, but after @timrozum and I talked about this, we agreed to not use masked characters here as this is a code that expires. Pretty common in apps like uber and lyft that ask you to enter a passcode via sms and it is not masked when entering. Not a huge deal though as it's a one liner 😄

Reply:
Yeah thoughts were:

1. This is a temporary passcode not a password.
2. It seems common for temporary passcodes to not use secure text box however not all passcodes are created equally.
3. It's not as inconvenient to use secure text box for password because users are used to typing password frequently or they have password manager.

But yeah easy to change if we want to.

Reply:
Other apps use OTPs to augment a user's existing credentials, so it makes sense not to mask the input. We are asking the user to use this passcode as their credentials. I wouldn't consider the password temporary since it can be used for longer than a few minutes. It would be considered temporary if users used the passcode to create their actual credentials, but that is not the case.

Let's use `numberPassword` 😄